libs/openFrameworks/graphics/ofTrueTypeFont.cpp

```diff
@@ -1054,12 +1054,16 @@ void ofTrueTypeFont::iterateString(const string & str, float x, float y, bool vF
 			} else if (c == '\t') {
 				if (settings.direction == OF_TTF_LEFT_TO_RIGHT){
 					f(c, pos);
-					pos.x += (getGlyphProperties(' ').advance * spaceSize * letterSpacing) * TAB_WIDTH * directionX;
+					pos.x += getGlyphProperties(' ').advance * letterSpacing * TAB_WIDTH * directionX;
 				} else{
-					pos.x += (getGlyphProperties(' ').advance * spaceSize * letterSpacing) * TAB_WIDTH * directionX;
+					pos.x += getGlyphProperties(' ').advance * letterSpacing * TAB_WIDTH * directionX;
 					f(c, pos);
 				}
 				prevC = c;
+			} else if(c == ' '){
+				pos.x += getGlyphProperties(' ').advance * spaceSize * TAB_WIDTH * directionX;
```

Comment:
I think TAB_WITH is not correct here? this is only to set the number of spaces per tab so this would add 4 spaces rather than 1 no?

Also i think we should just use spaceSize if it's set instead of advance * spaceSize otherwise it should be callsed space size factor instead of space size to avoid confusions

Reply:
No i think it's correct - this branch deals with handling the '\t' tab character, which we expand to 4 (or however many we specify in `TAB_WIDTH`) spaces, unless I'm mistaken.

Reply:
Looking at it again. It looks like TAB_WIDTH is unnecessary because the if statement above it should handle that case.  
`pos.x += getGlyphProperties(' ').advance * spaceSize * directionX;`

The only issue with removing getGlyphProperties(' ').advance is that then the default spacing is too tight.

`pos.x += spaceSize * directionX;`

It only includes a letter space. 
<img width="239" alt="Screen Shot 2019-04-23 at 2 15 53 PM" src="https://user-images.githubusercontent.com/3160465/56605655-9e9ae680-65d2-11e9-89cf-41de12fc4777.png">

So I'd think it would be best to leave it in.

Reply:
yes of course - i was looking at lines 1057, and 1059, sorry. the `TAB_WIDTH` in 1064 looks indeed misplaced.

Reply:
@crecord, yes i'm not saying we should remove the advance for the space just that if the space size is set then and only then instead of using the advance we should use the space size.

We can keep track with a flag (`bool spaceSizeSet`) if that value has been set or something like that

Not sure if that makes more sense but if not then that function should be called something like setSpaceSizeFactor or something similar

Reply:
I see, I think the factor approach would be preferable because then the spacing would scale with the text rather than needing to set it for a new text size.  Isn't letterSpacing also a factor, but that isn't indicated in the name? Seems like that should be consistent in the naming of both whichever direction you want to go.